images/router/haproxy/conf/haproxy-config.template

```diff
@@ -252,7 +252,13 @@ backend be_tcp_{{$cfgIdx}}
 {{ if ne (env "ROUTER_SYSLOG_ADDRESS" "") ""}}
   option tcplog
 {{ end }}
+    {{ with $balanceAlgo := index $cfg.Annotations "router.openshift.io/haproxy.balance" }}
+      {{ with $matchValue := (matchValues $balanceAlgo "roundrobin" "leastconn" ) }}
+  balance {{ $balanceAlgo }}
```

Comment:
No, the template processing does not fail. It just means that the 'with' block will not be processed further. Treat that as an 'if' statement.